Create a scrolling background and move the player

louissalin opened the pull request.

louissalin committed:
wip. Got a scrolling background but made some new changes so it doesn't compile right now

louissalin committed:
got a scrolling background strip

louissalin committed:
got scrolling of a background that fills the entire screen

louissalin committed:
add link to doc for understanding origin

louissalin committed:
use destinationRectangle instead of origin parameter when drawing to tile the terrain texture. Much easier to understand

louissalin committed:
remove commented code

louissalin committed:
put argument back in

louissalin committed:
remove unneeded using statement

louissalin committed:
remove unneeded using statement

louissalin committed:
add space to make the file be the same as origin

louissalin committed:
initialize the gameplay state by putting the player at the bottom in the middle

louissalin committed:
move player left and right and keep it in bounds

jcapellman commented on chapter-06/start/States/GameplayState.cs:
```diff
@@ -26,6 +37,33 @@ public override void HandleInput()
+
+        private void KeepPlayerInBounds()
+        {
+            if (_playerSprite.Position.X < 0)
```
Brace use for the conditionals (even single line) - I believe I had established that earlier

louissalin replied:
I have noticed your braced single line bodies. For some reason it didn't compute in my head that I should follow that style. Fixing.

louissalin replied:
fixed

jcapellman commented on chapter-06/start/Objects/Base/BaseGameObject.cs:
```diff
@@ -11,11 +11,18 @@ public class BaseGameObject
+        public int Height { get { return _texture.Height; } }
+        public Vector2 Position 
+        { 
+            get { return _position; } 
```
Expression Body here is style++ :)

louissalin replied:
eh, having the get and the set on the same line was just ugly

louissalin committed:
braces on single line bodies

louissalin commented:
you know I almost got to use pattern matching last night. I had added an event type and a pattern match before realizing, unfortunately, that I didn't need it. :(

oh, pattern matching will come! Just not yet...

louissalin committed:
Move dev to wip folder. Delete end folder.

I mistakenly worked on the start project instead of the end project, so I'm renaming it to wip and deleting the end project. Start and End will be rebuilt after I'm done with this chapter.
